Error on macro fragment

Error while parsing macro fragment

I tried this code:

```rust
macro_rules! foo {
    ($p:path $b:block) => {};
}

fn main() {}
```

I expected to see this happen: Compiles with unused macro warning.

Instead, this happened: Error

```
test.rs:2:14: error: fragment is not allowed after ‘path’ fragment
    2 |     ($p:path $b:block) => {};
      |              ^
test.rs:2:14: note: allowed tokens are `=>`, `,`, `=`, `|`, `;`, `:`, `>`, `>>`, `[`, `{`, `as`, `where`
test.rs:2:22: error: required first macro rule in macro rules definition could not be parsed
    2 |     ($p:path $b:block) => {};
      |                      ^
test.rs:2:22: error: failed to parse item in crate
```

### Meta

007166c

--- a/gcc/rust/ast/rust-macro.h
+++ b/gcc/rust/ast/rust-macro.h
@@ -149,6 +149,7 @@ public:
   {
     switch (kind)
       {
+      case PATH:
       case PAT:
       case TY:
       case VIS:

--- a/gcc/testsuite/rust/compile/macro-issue2653.rs
+++ b/gcc/testsuite/rust/compile/macro-issue2653.rs
@@ -1,0 +1,5 @@
+macro_rules! foo {
+    ($p:path $b:block) => {};
+}
+
+fn main() {}

